core/domain/interaction_jobs_one_off_test.py

```diff
@@ -874,3 +874,124 @@ def test_no_action_is_performed_for_deleted_exploration(self):
             self,
             interaction_jobs_one_off
             .InteractionCustomizationArgsValidationOneOffJob)
+
+
+class ContinueLabelValidationOneOffJobTests(test_utils.GenericTestBase):
+
+    ALBERT_EMAIL = 'albert@example.com'
+    ALBERT_NAME = 'albert'
+
+    VALID_EXP_ID = 'exp_id0'
+    NEW_EXP_ID = 'exp_id1'
+    EXP_TITLE = 'title'
+
+    def setUp(self):
+        super(ContinueLabelValidationOneOffJobTests, self).setUp()
+
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        # Setup user who will own the test explorations.
+        self.signup(self.ALBERT_EMAIL, self.ALBERT_NAME)
+        self.albert_id = self.get_user_id_from_email(self.ALBERT_EMAIL)
+        self.process_and_flush_pending_tasks()
+
+    def test_exp_state_pairs_are_produced_only_for_desired_interactions(self):
+        """Checks output pairs are produced only for desired interactions."""
+
+        exploration = exp_domain.Exploration.create_default_exploration(
+            self.VALID_EXP_ID, title='title', category='category')
+
+        exploration.add_states(['State1', 'State2'])
+
+        state1 = exploration.states['State1']
+        state2 = exploration.states['State2']
+
+        state1.update_interaction_id('Continue')
+
+        customization_args_dict1 = {
+            'buttonText': {
+                'value': {
+                    'content_id': 'CONTENT_ID',
+                    'unicode_str': 'a' * 50
+                }
+            }
+        }
+
+        state1.update_interaction_customization_args(customization_args_dict1)
+        exp_services.save_new_exploration(self.albert_id, exploration)
+
+        # Start ItemSelectionInteractionOneOff job on sample exploration.
+        job_id = (
+            interaction_jobs_one_off
+            .InteractionCustomizationArgsValidationOneOffJob.create_new())
```

Comment:
WHy are we running `InteractionCustomizationArgsValidationOneOffJob` job? [Ditto elsewhere in this file!]

Reply:
MB, I've fixed it now.